docs/specs/clients/notify/client-sdk-api.md

```diff
@@ -56,17 +58,33 @@ abstract class Client {
     encryptedMessage: string,
   }): Promise<NotifyMessage>;
   
+  // Generates and returns SIWE message along with
+  // Cacao Payload with identity key
+  public abstract prepareRegistration(params: {
+    account: string;
+    domain: string,
+    // Defaulted to false if not passed
+    allApps?: boolean,
+  }): {
+    registerParams: NotifyRegistrationParams,
+    message: string
+  };
+  
   // "Logs in" this client to notifications for the specified account. This involves:
-  // - Generating, signing, saving to local stroage, and registering an Identity Key, if one didn't already exist in local storage
+  // - Verifies signature by reconstructing formatted message using provided payload
+  // - Expected payload comes from prepareRegistration
+  // - Once verified, registers the identity key on the keyserver
   // - Calls `wc_notifyWatchSubscriptions`, watches for future subscription updates, and subscribes to all notification topics
   // Returns the Identity Key public key. Method should throw 'signatureRejected' if any errors comes from onSign promise. 
   public abstract register(params: {
-    account: string;
-    private?: boolean;
-    domain: string,
-    isLimited?: boolean,
-    onSign: (message: string) => Promise<Cacao.Signature>
+    registerParams: NotifyRegistrationParams,
+    signature: string;
   }): Promise<string>;
+  
+  // Returns true if account has a valid and up to date registration, false otherwise
```

Comment:
```suggestion
  // Returns true if account has a valid and up to date registration, false otherwise
  // If false, the client can still receive notifications from this account, but any actions (e.g. updating a subscription or creating a new one) will be rejected internally by the SDK
```

Reply:
I think making that distinction is over-engineering. If `isRegistered` is false they need to/should register. Adding an extra boolean state or making `isRegistered` return a struct will be a bit confusing

Reply:
As someone who will have to implement this into web3inbox, I have too many choices at this point. Let's assume we split the states. 

1. `isRegistered`: has a statement/identity key. Does not matter if it is stale
2. `hasStaleStatement`: has a stale statement

When do I show the signature modal on web3inbox? 
1. `!isRegistered`
2. `isRegistered && hasStaleStatement`

Because that it is effectively what people creating UIs will have to think about. What would we be recommending?